src/Collector/ProfileClient.php

```diff
@@ -26,56 +26,32 @@ class ProfileClient implements ClientInterface, HttpAsyncClient
 {
     use VersionBridgeClient;
 
-    /**
-     * @var ClientInterface|HttpAsyncClient
-     */
-    private $client;
+    private ClientInterface&HttpAsyncClient $client;
 
-    /**
-     * @var Collector
-     */
-    private $collector;
-
-    /**
-     * @var Formatter
-     */
-    private $formatter;
-
-    /**
-     * @var Stopwatch
-     */
-    private $stopwatch;
-
-    /**
-     * @var array
-     */
-    private $eventNames = [];
+    private array $eventNames = [];
 
     private const STOPWATCH_CATEGORY = 'httplug';
 
     /**
      * @param ClientInterface|HttpAsyncClient $client The client to profile. Client must implement HttpClient or
      *                                                HttpAsyncClient interface.
      */
-    public function __construct($client, Collector $collector, Formatter $formatter, Stopwatch $stopwatch)
+    public function __construct($client, private readonly Collector $collector, private readonly Formatter $formatter, private readonly Stopwatch $stopwatch)
```

Comment:
separate lines. and $client can be typehinted [since php 8.0](https://3v4l.org/Wnp4m)
```suggestion
    public function __construct(
        ClientInterface|HttpAsyncClient $client, 
        private readonly Collector $collector, 
        private readonly Formatter $formatter, 
        private readonly Stopwatch $stopwatch,
    )
```

Reply:
++